create.tf

```diff
@@ -1,40 +1,110 @@
-// Create Database will create a database
-// Additionally, a role of the same name will be created and given "owner" over database
-data "aws_lambda_invocation" "create-database" {
-  function_name = local.db_admin_func_name
-
-  input = jsonencode({
-    type = "create-database"
-    metadata = {
-      databaseName = local.database_name
-    }
-  })
-}
-
-data "aws_lambda_invocation" "create-user" {
-  function_name = local.db_admin_func_name
-
-  input = jsonencode({
-    type = "create-user"
-    metadata = {
-      username = local.username
-      password = random_password.this.result
-    }
-  })
-
-  depends_on = [data.aws_lambda_invocation.create-database]
-}
-
-data "aws_lambda_invocation" "create-db-access" {
-  function_name = local.db_admin_func_name
-
-  input = jsonencode({
-    type = "create-db-access"
-    metadata = {
-      databaseName = local.database_name
-      username     = local.username
-    }
-  })
-
-  depends_on = [data.aws_lambda_invocation.create-user]
-}
+provider "restapi" {
+  uri                  = coalesce(local.db_admin_func_url, "https://noop")
+  write_returns_object = true
+
+  aws_v4_signing {
+    service           = "lambda"
+    access_key_id     = try(local.db_admin_invoker["access_key"], null)
+    secret_access_key = try(local.db_admin_invoker["secret_key"], null)
```

Comment:
How does this key get passed? Is this secure?

Reply:
This gets passed by state file outputs. It's secure as long as we protect the state files.